arroyo-macro/src/lib.rs

```diff
@@ -472,6 +490,23 @@ fn impl_stream_node_type(
             }
         }
     } else {
+        let tick_setup = tick_ms.as_ref().map(|t| {
+            quote! {
+                let mut ticks = 0u64;
+                let mut next_tick = std::time::SystemTime::now() + std::time::Duration::from_millis(#t);
+            }
+        });
+
+        let tick_case = tick_ms.as_ref().map(|t| {
+            quote! {
+                _ = tokio::time::sleep(next_tick.duration_since(SystemTime::now()).unwrap_or(Duration::ZERO)) => {
```

Comment:
What about using `tokio::time::interval`? Something like

```
let mut interval = tokio::time::interval([Duration](std::time::Duration::from_millis(#t)));
interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
```
and then
```
_ = interval.tick() {}
```

in the select block.

Reply:
nice, didn't know about tokio::interval